Firestore/core/src/remote/bloom_filter.cc

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+
+#include "Firestore/core/src/remote/bloom_filter.h"
+
+#include "CommonCrypto/CommonDigest.h"
+#include "Firestore/core/src/util/hard_assert.h"
+#include "Firestore/core/src/util/log.h"
+#include "Firestore/core/src/util/statusor.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using util::Status;
+using util::StatusOr;
+
+/** Helper function to hash a string using md5 hashing algorithm, and return an
+ * array of 16 bytes. */
+BloomFilter::Hash BloomFilter::Md5HashDigest(
+    const absl::string_view key) const {
+  unsigned char md5_digest[CC_MD5_DIGEST_LENGTH];
+
+  CC_MD5_CTX context;
+  CC_MD5_Init(&context);
+  CC_MD5_Update(&context, key.data(), key.size());
+  CC_MD5_Final(md5_digest, &context);
+
+  uint64_t* hash128 = reinterpret_cast<uint64_t*>(md5_digest);
```

Comment:
This `reinterpret_cast` may not be "safe" here. It is safe on the server because we know that the code will be running on linux x86_64 architecture. I think we should be safer and avoid the `reinterpret_cast` because it is really flirting with undefined behavior.

Let's leave it as-is for now, but please add a TaskFlow to look into replacing this logic with something like Android's `getLongLittleEndian()` method that you wrote: https://github.com/firebase/firebase-android-sdk/blob/9e7023d9d50e8a5c3fbf8b492bb4dd51e44da87c/firebase-firestore/src/main/java/com/google/firebase/firestore/remote/BloomFilter.java#L110-L116

Reply:
Please add a TODO comment in the code too, referencing the Buganizer ticket that you create.